language-plutus-core/src/Language/PlutusCore/Pretty/Plc.hs

```diff
@@ -0,0 +1,118 @@
+-- | The global pretty-printing config used to pretty-print everything in the PLC world.
+-- This module also defines custom pretty-printing functions for PLC types as a convenience.
+
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+{-# LANGUAGE ConstraintKinds       #-}
+{-# LANGUAGE FlexibleInstances     #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE TypeFamilies          #-}
+{-# LANGUAGE UndecidableInstances  #-}
+
+module Language.PlutusCore.Pretty.Plc
+    (
+    -- * Global configuration
+      PrettyConfigPlc (..)
+    , PrettyPlc
+    , defPrettyConfigPlcClassic
+    , debugPrettyConfigPlcClassic
+    , defPrettyConfigPlcRefined
+    , debugPrettyConfigPlcRefined
+    -- * Custom functions for PLC types.
+    , prettyPlcClassicBy
+    , prettyPlcClassicDefBy
+    , prettyPlcClassicDef
+    , prettyPlcClassicDebug
+    , prettyPlcRefinedBy
+    , prettyPlcRefinedDefBy
+    , prettyPlcRefinedDef
+    , prettyPlcRefinedDebug
+    ) where
+
+import           Language.PlutusCore.Pretty.Classic
+import           Language.PlutusCore.Pretty.Name
+import           Language.PlutusCore.Pretty.Refined
+import           Language.PlutusCore.Type
+import           PlutusPrelude
+
+-- | Global configuration used for pretty-printing PLC entities.
+data PrettyConfigPlc
+    = PrettyConfigPlcClassic (PrettyConfigClassic PrettyConfigName)
+    | PrettyConfigPlcRefined (PrettyConfigRefined PrettyConfigName)
+
+-- | The "pretty-printable PLC entity" constraint.
+type PrettyPlc = PrettyBy PrettyConfigPlc
+
+-- | A constraint that allows to derive @PrettyBy PrettyConfigPlc@ instances, see below.
+type DefaultPrettyPlc a =
+       ( PrettyBy (PrettyConfigClassic PrettyConfigName) a
+       , PrettyBy (PrettyConfigRefined PrettyConfigName) a
+       )
+
+instance configName ~ PrettyConfigName => HasPrettyConfigName (PrettyConfigClassic configName) where
```

Comment:
Because this implies that other `HasPrettyConfigName (PrettyConfigClassic Something)` instances can be added which is not the intention. The current version also plays better with type inference. The trick is described [here](https://chrisdone.com/posts/haskell-constraint-trick).